web/template/create_document.gohtml

```diff
@@ -5,204 +5,155 @@
-                                            <div class="govuk-checkboxes" data-module="govuk-checkboxes">
-                                                <div class="govuk-checkboxes__item">
-                                                    <input class="govuk-checkboxes__input" id="f-correspondenceBy-post" name="correspondenceBy" type="checkbox" value="post">
-                                                    <label class="govuk-label govuk-checkboxes__label" for="f-correspondenceBy-post">
-                                                        Post
-                                                    </label>
-                                                </div>
-                                                <div class="govuk-checkboxes__item">
-                                                    <input class="govuk-checkboxes__input" id="f-correspondenceBy-email" name="correspondenceBy" type="checkbox" value="email">
-                                                    <label class="govuk-label govuk-checkboxes__label" for="f-correspondenceBy-email">
-                                                        Email
-                                                    </label>
-                                                </div>
-                                                <div class="govuk-checkboxes__item">
-                                                    <input class="govuk-checkboxes__input" id="f-correspondenceBy-phone" name="correspondenceBy" type="checkbox" value="phone">
-                                                    <label class="govuk-label govuk-checkboxes__label" for="f-correspondenceBy-phone">
-                                                        Phone
-                                                    </label>
-                                                </div>
-                                                <div class="govuk-checkboxes__item">
-                                                    <input class="govuk-checkboxes__input" id="f-correspondenceBy-welsh" name="correspondenceBy" type="checkbox" value="welsh">
-                                                    <label class="govuk-label govuk-checkboxes__label" for="f-correspondenceBy-welsh">
-                                                        Welsh
-                                                    </label>
-                                                </div>
+            {{ if .HasSelectedAddNewRecipient }}
+                <form class="form" method="GET">
+                    {{ $url := (printf "create-document?id=%d&case=%s&templateId=%s&hasViewedInserts=true" .Case.ID .Case.CaseType .TemplateSelected.TemplateId) }}
+                    {{ range .SelectedInserts }}
+                        {{ $url = (printf "%s&insert=%s" $url .)  }}
+                    {{ end }}
+                    <a href="{{$url}}" class="govuk-back-link">Back</a>
```

Comment:
couldn't you do

```
{{ range .SelectedInserts }}
  <input type="hidden" name="selectedInserts" value="{{ . }}" />
{{ end }}
```

to use hidden inputs for these?

Reply:
not that I'm totally against keeping the state in the URL, for your template function you probably needed to return `template.HTML("whatever string")` instead of a string, then it won't escape it

Reply:
Using hidden inputs isn't an option as it's a link element. I've given returning a` template.HTML` string a go and it still escapes the `=` and `&` so for now i have kept this as is